tt_metal/impl/dispatch/kernels/cq_dispatch.cpp

```diff
@@ -74,13 +74,20 @@ static uint32_t cmd_ptr;   // walks through pages in cb cmd by cmd
 static uint32_t downstream_cb_data_ptr = downstream_cb_base;
 
 constexpr uint32_t l1_to_local_cache_copy_chunk = 6;
-constexpr uint32_t max_write_packed_cores =
-    108;  // GS 120 - 1 row TODO: this should be a compile time arg passed in from host
-constexpr uint32_t l1_cache_size =
-    ((max_write_packed_cores + l1_to_local_cache_copy_chunk - 1) / l1_to_local_cache_copy_chunk) *
+
+// For packed write on GS, below must be>= 120 - 1. TODO: this should be a compile time arg passed in from host
+// For packed_write_large, must match CQ_DISPATCH_CMD_PACKED_WRITE_LARGE_MAX_SUB_CMDS
+constexpr uint32_t max_write_packed_cores = 108;
+constexpr uint32_t max_write_packed_large_cmd =
+    CQ_DISPATCH_CMD_PACKED_WRITE_LARGE_MAX_SUB_CMDS *
+    sizeof(CQDispatchWritePackedLargeSubCmd) / sizeof(uint32_t);
+constexpr uint32_t l1_cache_elements = (max_write_packed_cores > max_write_packed_large_cmd) ?
+    max_write_packed_cores : max_write_packed_large_cmd;
```

Comment:
I don't fully understand whats happening here. Where is the value for `max_write_packed_cores ` coming from? This is what we have for a 1 row harvested Grayskull.
It seems like the `l1_cache_elements` are computed based on how many subcmds they need to store (copied using `careful_copy_from_l1_to_local_cache`)?

Reply:
there are 2 commands that use the buffer, the code above is calculating the size each one needs and picking the larger.  I could have done that manually and then just asserted that each type was <= to the manual value, that may be clearer, lmk